plugins/tedge_configuration_plugin/src/main.rs

```diff
@@ -0,0 +1,135 @@
+after_help = AFTER_HELP_TEXT
+)]
+pub struct ConfigPluginOpt {
+    /// Turn-on the debug log level.
+    ///
+    /// If off only reports ERROR, WARN, and INFO
+    /// If on also reports DEBUG
+    #[clap(long)]
+    pub debug: bool,
+
+    #[clap(long = "config-dir", default_value = DEFAULT_TEDGE_CONFIG_PATH)]
+    pub config_dir: PathBuf,
+
+    #[clap(long)]
+    mqtt_topic_root: Option<Arc<str>>,
+
+    #[clap(long)]
+    mqtt_device_topic_id: Option<Arc<str>>,
+}
+
+#[tokio::main]
+async fn main() -> Result<(), anyhow::Error> {
+    let config_opt = ConfigPluginOpt::parse();
+
+    // Load tedge config from the provided location
+    let tedge_config_location = TEdgeConfigLocation::from_custom_root(&config_opt.config_dir);
+
+    let log_level = if config_opt.debug {
+        tracing::Level::DEBUG
+    } else {
+        get_log_level(
+            "tedge-configuration-plugin",
+            &tedge_config_location.tedge_config_root_path,
+        )?
+    };
+    set_log_level(log_level);
+
+    let tedge_config = TEdgeConfigRepository::new(tedge_config_location).load()?;
+
+    run(tedge_config, config_opt).await
+}
+
+async fn run(tedge_config: TEdgeConfig, cliopts: ConfigPluginOpt) -> Result<(), anyhow::Error> {
+    let runtime_events_logger = None;
+    let mut runtime = Runtime::try_new(runtime_events_logger).await?;
+
+    let mqtt_topic_root = cliopts
+        .mqtt_topic_root
+        .unwrap_or(tedge_config.mqtt.topic_root.clone().into());
+
+    let mqtt_device_topic_id = cliopts
+        .mqtt_device_topic_id
+        .unwrap_or(tedge_config.mqtt.device_topic_id.clone().into());
+
+    let mqtt_config = tedge_config.mqtt_config()?;
+    let mut mqtt_actor = MqttActorBuilder::new(mqtt_config.clone().with_session_name(format!(
+        "{TEDGE_CONFIGURATION_PLUGIN}#{mqtt_topic_root}/{mqtt_device_topic_id}",
+    )));
+
+    let mut fs_watch_actor = FsWatchActorBuilder::new();
+
+    let health_actor = HealthMonitorBuilder::new(TEDGE_CONFIGURATION_PLUGIN, &mut mqtt_actor);
+
+    let mut http_actor = HttpActor::new().builder();
+
+    let mut downloader_actor = DownloaderActor::new().builder();
+
+    // Instantiate config manager actor
+    let manager_config = ConfigManagerConfig::from_options(ConfigManagerOptions {
+        config_dir: cliopts.config_dir,
+        mqtt_device_topic_id,
+        mqtt_topic_root,
+    })?;
+
+    let config_actor = ConfigManagerBuilder::try_new(
+        manager_config,
+        &mut mqtt_actor,
+        &mut http_actor,
+        &mut fs_watch_actor,
+        &mut downloader_actor,
+    )?;
+
+    // Shutdown on SIGINT
+    let signal_actor = SignalActor::builder(&runtime.get_handle());
+
+    // Run the actors
+    runtime.spawn(mqtt_actor).await?;
+    runtime.spawn(http_actor).await?;
+    runtime.spawn(downloader_actor).await?;
+    runtime.spawn(fs_watch_actor).await?;
+    runtime.spawn(config_actor).await?;
+    runtime.spawn(signal_actor).await?;
+    runtime.spawn(health_actor).await?;
```

Comment:
`downloader_actor` should be spawaned!

Reply:
I've updated the config plugin with that fix just before you posted this comment.